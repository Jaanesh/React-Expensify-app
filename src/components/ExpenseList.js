import React from 'react';
import {connect} from 'react-redux' ;
import listFilteredExpenses from '../selectors/Expenses';
import ExpenseItem from './ExpenseListItemComponent';
import ExpenseListFilterComponent from './ExpenseListFilterComponent';


export const ExpenseList=(props)=>{

   // let filteredExpenses=listFilteredExpenses(props.expenses,props.filters);

    return (
    <div>
        <ExpenseListFilterComponent></ExpenseListFilterComponent>
        <p>Expense List</p>
        <table>
       
            <tbody>
            <tr>
               <th>description</th>
               <th>amount</th> 
               <th>createdAt</th>
           </tr>
           {
               
         /*   filteredExpenses.map((eachExpense,index)=>{
                                 return (<ExpenseItem key={eachExpense.id} {...eachExpense}></ExpenseItem>);
                               }
            )*/
            props.expenses.map((eachExpense,index)=>{                                  
                                return (<ExpenseItem key={eachExpense.id} {...eachExpense}></ExpenseItem>);
                              }  )          
                         }
            </tbody>
        
       </table>
       

    </div>
)};

/*const ConnectedExpenseListComponent=connect((state)=>{
    return {
        expenses:state.expenses
    }
})(ExpenseList);

export default ConnectedExpenseListComponent;
*/

const mapStateToProps=(state)=>{
  /*  return {
        expenses:state.expenses,
        filters:state.filters
    }*/

   return{
        //list the expenses based on filter 
        expenses:listFilteredExpenses(state.expenses,state.filters)
    }
   
}

export default connect(mapStateToProps)(ExpenseList);

