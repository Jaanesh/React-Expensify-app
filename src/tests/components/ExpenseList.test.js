import React from 'react';
import {shallow} from 'enzyme';
import moment from 'moment';
//import 'react-dates/lib/css/_datepicker.css';
import {ExpenseList} from '../../components/ExpenseList.js';

let expenseObj1={id:'1', description:'jan rent',note:'',amount:3,createdAt:moment(0).
                                                                           subtract(5,'days').valueOf()};
let expenseObj2={id:'2', description:'feb rent',note:'',amount:2,createdAt:moment(0)};
let expenseObj3={id:'3', description:'mar fees',note:'',amount:1,createdAt:moment(0).
                                                                           add(5,'days').valueOf()};

let expensesArray=[expenseObj1,expenseObj2,expenseObj3];


test('should render ExpenseList with expenses',()=>{
    const wrapper=shallow(<ExpenseList expenses={expensesArray} />);
    expect(wrapper).toMatchSnapshot();
});

test('should render expense list with no expense',()=>{
    const wrapper=shallow(<ExpenseList expenses={[]} />);
    expect(wrapper).toMatchSnapshot();
});