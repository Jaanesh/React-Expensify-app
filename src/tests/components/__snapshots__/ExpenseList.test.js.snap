// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseList with expenses 1`] = `
<div>
  <Connect(ExpenseListFilterComponent) />
  <p>
    Expense List
  </p>
  <table>
    <tbody>
      <tr>
        <th>
          description
        </th>
        <th>
          amount
        </th>
        <th>
          createdAt
        </th>
      </tr>
      <Component
        amount={3}
        createdAt={-432000000}
        description="jan rent"
        id="1"
        key="1"
        note=""
      />
      <Component
        amount={2}
        createdAt={"1970-01-01T00:00:00.000Z"}
        description="feb rent"
        id="2"
        key="2"
        note=""
      />
      <Component
        amount={1}
        createdAt={432000000}
        description="mar fees"
        id="3"
        key="3"
        note=""
      />
    </tbody>
  </table>
</div>
`;

exports[`should render expense list with no expense 1`] = `
<div>
  <Connect(ExpenseListFilterComponent) />
  <p>
    Expense List
  </p>
  <table>
    <tbody>
      <tr>
        <th>
          description
        </th>
        <th>
          amount
        </th>
        <th>
          createdAt
        </th>
      </tr>
    </tbody>
  </table>
</div>
`;
